pkg/kv/dynamodb/store.go

```diff
@@ -376,6 +379,14 @@ func (s *Store) scanInternal(ctx context.Context, partitionKey, scanKey []byte,
 	}, nil
 }
 
```

Comment:
alternative - if it is awserr have a struct with this error embed or another type. implement the go standard Unwrap signature that will return the original error.
This will remove the cancel error check here and support other errors we like to check using errors.Is.

Reply:
`awserr.RequestError` doesn't have `Unwrap`, just `OrigErr`

Reply:
Yes I know - we need to switch to the new Go AWS SDK :/
What I had in mind to make the handle more generic by:

```go
type AWSError struct {
	Err awserr.Error
}

func (e *AWSError) Error() string {
	return e.Err.Error()
}

func (e *AWSError) Code() string {
	return e.Err.Code()
}

func (e *AWSError) Message() string {
	return e.Err.Message()
}

func (e *AWSError) OrigErr() error {
	return e.Err.OrigErr()
}

func (e *AWSError) Unwrap() error {
	return e.Err.OrigErr()
}

func handleClientError(err error) error {
	var awsError awserr.Error
	if errors.As(err, &awsError) {
		return &AWSError{awsError}
	}
	return err
}
```

Reply:
Do you suggest to wrap the client and convert all returned errors? Otherwise I can't think of a generic way which doesn't go thru wraping every returned error explicitly.